addons/web_editor/models/ir_ui_view.py

```diff
@@ -172,7 +172,8 @@ def _copy_field_terms_translations(self, record_from, name_field_from, record_to
             record_from[name_field_from],
             {lang: record_from.with_context(prefetch_langs=True, lang=lang)[name_field_from] for lang in langs if lang != lang_env}
         )
-        existing_translation_dictionary.update(extra_translation_dictionary)
+        if overwrite or not any(translated_value != key for key in extra_translation_dictionary.keys() for translated_value in existing_translation_dictionary[key].values()): 
+            existing_translation_dictionary.update(extra_translation_dictionary)
```

Comment:
I think it is not the logic `overwrite`

`overrwrite=False` means
for each `key` `value` pair defined in the `extra_translation_dictionary`, update the `existing_translation_dictionary` if
1.  the `key` is not in `existing_translation_dictionary`
2. or `existing_translation_dictionary[key] != key`

Also, could you add a test?